test/framework/debug.go

```diff
@@ -17,26 +17,36 @@ import (
 // * CP logs (although we print this already on failure)
 func DebugUniversal(cluster Cluster, mesh string) {
 	ensureDebugDir()
+	errorSeen := false
 	Logf("printing debug information of cluster %q for mesh %q", cluster.Name(), mesh)
 	// we don't have command to print policies for given mesh, so it's better to print all than none.
 	kumactlOpts := *cluster.GetKumactlOptions()
 	kumactlOpts.Verbose = false
 	out, err := kumactlOpts.RunKumactlAndGetOutput("export", "--profile", "all")
-	Expect(err).ToNot(HaveOccurred())
+	if err != nil {
+		// We don't want to fail in the middle.
+		out = fmt.Sprintf("kumactl export failed with error: %s", err)
+		errorSeen = true
+	}
 
 	exportFilePath := filepath.Join(Config.DebugDir, fmt.Sprintf("%s-export-%s", cluster.Name(), uuid.New().String()))
 	Expect(os.WriteFile(exportFilePath, []byte(out), 0o600)).To(Succeed())
+	Expect(errorSeen).To(BeTrue(), "some debug commands failed")
```

Comment:
Why expect it to be true? Shouldn't we just keep the error and expect it with `Not(HaveOccurred())`

Reply:
You are right fixed!